### DISCUSSION TITLE
Try to open GUI but only showing bash shells

### DISCUSSION THREAD

**Discussion by bessFX** [neurodesk/neurodesk.github.io-hugo#417] [Show and tell]
Dear experts,

I am new to NeuroDesk, and imaging analysis. I just downloaded the NeuroDesk app (version 1.4.0) on MacOS (Ventura 13.5). I wonder why GUIs didn’t showed up whenever I try to open one. For example, I tried to open SPM12GUI through the application menu, but only a bash shell showed up and no GUI appeared. What could caused the problem and how should I fixed it?

Sincere appreciating any helps in advance.

Best,
Xuan Feng

**Comment by stebo85** [comment]
Dear @bessFX,
What error do you get when you open a normal linux terminal and type `ml av`? You should see this:

<img width="1301" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/9abda5ba-a433-4ed7-bb95-c24f17fced16">


Can you try to open other GUIs? If you are running an Apple Silicon M1/M2/M3 and it only fails for SPM12, CAT12 and FSLeyes then you are seeing a bug in Docker in conjunction with Rosetta2. There are two ways around this:
1) Update Macos and Docker to the latest versions
2) Disable Rosetta in the Settings of Docker:
<img width="1430" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/c7d16923-6a07-46cb-90ee-7b0e649ed2f1">

**Reply by bessFX** [reply]
Dear Stebo,

Thank you so much for replying. Fortunately, I have no error when typing "ml av" on normal linux terminal in neurodesk app. I can open other GUIs, like mrviewGUI and dsistuidioGUI. I tried both methods, including updating MacOS and docker, and disabled Rosetta on docker setting, but I still can't open SPM, cat12 and FSLeyes. 
Is there any other way to fix this?

This what I got after click SPM12GUI from the application menu:
(somehow I can't upload pictures)
the error I got from the bash shell that opened up:
Can't detect primordial thread stack location - find_vma failed

How should I fix the error? Thank you again for your help: )

Best,
Xuan

**Reply by bessFX** [reply]
![Uploading Screenshot 2024-03-28 at 16.29.54.png…]()

**Reply by stebo85** [reply]
Are you running on Apple silicon m1/m2/m3 ?

**Reply by bessFX** [reply]
Yes, I am running on Apple silicon m2.

**Reply by stebo85** [reply]
Can you confirm that you are running:
MacOS Sonoma 14.4.1 (23E224)
Docker Desktop 4.28.0 (139021)

Older versions will have this bug in Rosetta 2.

Another option is to update the Neurodeskapp to 1.5.0, install Podman, and then switch to Podman in the settings:
<img width="707" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/f6b11e68-1d67-47ec-9363-c7ba6a78d88c">

**Reply by scott-love** [reply]
Hi @stebo85 and @bessFX,

I was having a problem with the SPM GUI, which led me to find this post. Basically the initial GUI would appear but a few seconds after clicking on the fMRI option everything would crash - including the terminal. 
However, thanks to this post, and in particular the part with a helpful image about the Rosetta docker settings, I disabled Rosetta and was able to open the SPM-fMRI GUI. 
That said, I did see the same message in the terminal "Can't detect primordial thread stack location - find_vma failed" as @bessFX. However, a little patience (it does take a few minutes) and the GUI still appears. 

So in my case disabling Rosetta enabled me to use the SPM GUI.

I haven't tried Podman, yet. 

My setup:
MacOS Sonoma 14.4.1 (23E224)
Docker Desktop 4.28.0 (139021)
Apple M3 Pro

**Reply by stebo85** [reply]
Dear @scott-love,

Yes, I get the same behaviour with Rosetta 2 as well: The SPM GUI starts, but it crashes indeed when clicking on fMRI. This is unfortunately yet another Rosetta bug and we need to wait for Apple to fix that. The only workaround for now is to disable Rosetta 2 in Docker or use Podman (which should both give the same result). Unfortunately, this means that everything is about 5 times slower.

The best long-term fix would be to build an arm64 version of SPM - but the problem is that Mathworks doesn't provide an arm64 version of the Matlab runtime environment yet.
